src/lib/index.ts

```diff
@@ -66,7 +66,8 @@ class ServerlessIamPerFunctionPlugin {
     let length=0; //calculate the expected length. Sum the length of each part
     for (const part of name_parts) {
       if (part.Ref) {
-        length += part.Ref.length;
+        const ref = this.resolveRef(part.Ref)
```

Comment:
Could you simply replace this with
```javascript
length += this.serverless.service.provider.region
```
This would resolve the reduction in coverage